jbmc/src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -1237,12 +1240,51 @@ codet java_bytecode_convert_methodt::convert_instructions(
     else if(statement=="athrow")
     {
       PRECONDITION(op.size() == 1 && results.size() == 1);
+      if(
+        !throw_assertion_error &&
+        uncaught_exceptions_domaint::get_exception_type(op[0].type()) ==
+          "java::java.lang.AssertionError")
+      {
+        // we translate athrow into
+        // __CPROVER_assert(false, comment);
+        // __CPROVER_assume(false);
+        exprt comment = address_of_exprt(index_exprt(
+          string_constantt("assertion at " + i_it->source_location.as_string()),
+          from_integer(0, index_type())));
+        code_typet assert_type(
+          {code_typet::parametert(bool_typet()),
+           code_typet::parametert(comment.type())},
+          void_typet());
+        code_function_callt assert_call;
+        assert_call.function() =
+          symbol_exprt(CPROVER_PREFIX "assert", assert_type);
+        assert_call.lhs().make_nil();
+        assert_call.arguments().push_back(false_exprt());
+        assert_call.arguments().push_back(std::move(comment));
+        assert_call.add_source_location() = i_it->source_location;
+
+        code_typet assume_type(
+          {code_typet::parametert(bool_typet())}, void_typet());
+        code_function_callt assume_call;
+        assume_call.function() =
+          symbol_exprt(CPROVER_PREFIX "assume", assume_type);
+        assume_call.lhs().make_nil();
+        assume_call.arguments().push_back(false_exprt());
+        assume_call.add_source_location() = i_it->source_location;
 
-      side_effect_expr_throwt throw_expr;
-      throw_expr.add_source_location()=i_it->source_location;
-      throw_expr.copy_to_operands(op[0]);
-      c=code_expressiont(throw_expr);
-      results[0]=op[0];
+        code_blockt ret_block;
+        ret_block.operands().push_back(std::move(assert_call));
```

Comment:
Unsure which we prefer of `std::move` vs. `move_to_operands` these days, but check